app/queries/draft_and_live_versions.rb

```diff
@@ -0,0 +1,65 @@
+module Queries
+  module DraftAndLiveVersions
+    extend ArelHelpers
+    def self.call(content_item, target_table_name, locale = nil)
+      content_items_table = table(:content_items)
+      states_table = table(:states)
+      translations_table = table(:translations)
+      versions_table = table(target_table_name)
+      versions_fk = target_table_name == "lock_versions" ? :target_id : :content_item_id
+
+      if locale.nil?
+        locale = translations_table.project(translations_table[:locale])
+          .where(translations_table[:content_item_id].eq(content_item.id))
+      end
+
+      scope = content_items_table
+        .project(
+          Arel::Nodes::SqlLiteral.new(
+            "CASE WHEN states.name IN ('published', 'unpublished') THEN 'live' " +
+            "WHEN states.name = 'draft' THEN 'draft' END"
+          ).as("state"),
+          versions_table[:number].as("version_number")
+        )
+        .join(states_table).on(
+          content_items_table[:id].eq(states_table[:content_item_id])
+        )
+        .join(translations_table).on(
+          content_items_table[:id].eq(translations_table[:content_item_id])
+        )
+        .join(versions_table).on(
+          content_items_table[:id].eq(versions_table[versions_fk])
+        )
+        .where(content_items_table[:content_id].eq(content_item.content_id))
+        .where(translations_table[:locale].eq(locale))
+        .where(states_table[:name].in(%w(draft published unpublished)))
+      if target_table_name == "lock_versions"
+        scope.where(versions_table[:target_type].eq("ContentItem"))
```

Comment:
I'm not sure how this works, as scope is not being assigned (it must be a mutation, as I tested it, and it works...).